File: social-pipelines/patreon/scripts/x_post.js
#!/usr/bin/env node
/**
 * X (Twitter) API v2 Auto Poster
 * Usage:
 *   node x_post.js "message"
 *   node x_post.js "message" /path/to/image.jpg
 */

const { TwitterApi } = require('twitter-api-v2');
require('dotenv').config();

const client = new TwitterApi({
  appKey: process.env.X_API_KEY,
  appSecret: process.env.X_API_SECRET,
  accessToken: process.env.X_ACCESS_TOKEN,
  accessSecret: process.env.X_ACCESS_SECRET,
});

const rw = client.readWrite;

async function postTweet(text, imagePath) {
  console.log('🐦 X API v2 Auto Poster');
  console.log(`   Text: ${text.length} chars`);
  console.log(`   Image: ${imagePath || 'none'}`);

  try {
    let mediaId = null;

    if (imagePath) {
      console.log('📸 Uploading image...');
      const media = await rw.v1.uploadMedia(imagePath);
      mediaId = media.media_id_string;
      console.log(`   Media ID: ${mediaId}`);
    }

    console.log('📝 Posting tweet...');
    const tweetParams = { text };
    if (mediaId) {
      tweetParams.media = { media_ids: [mediaId] };
    }

    const tweet = await rw.v2.tweet(tweetParams);
    const tweetId = tweet.data.id;
    console.log(`✅ Tweet posted!`);
    console.log(`   https://x.com/VotiveAstrology/status/${tweetId}`);
    return tweetId;
  } catch (err) {
    console.error(`❌ Error: ${err.message}`);
    if (err.data) {
      console.error(`   Detail: ${JSON.stringify(err.data).slice(0, 500)}`);
    }
    process.exit(1);
  }
}

// Parse args
const args = process.argv.slice(2);
if (args.length < 1) {
  console.log('Usage: node x_post.js "message" [image_path]');
  process.exit(1);
}

const text = args[0];
const imagePath = args[1] || null;

postTweet(text, imagePath);

File: social-pipelines/patreon/scripts/x_post.py
#!/usr/bin/env python3
"""
X (Twitter) API v2 Auto Poster
Usage:
  python3 x_post.py "message"
  python3 x_post.py "message" /path/to/image.jpg
"""

import tweepy
import json
import sys
import os
from dotenv import load_dotenv

load_dotenv()

def get_client():
    return tweepy.Client(
        consumer_key=os.environ.get("X_API_KEY"),
        consumer_secret=os.environ.get("X_API_SECRET"),
        access_token=os.environ.get("X_ACCESS_TOKEN"),
        access_token_secret=os.environ.get("X_ACCESS_SECRET"),
    )

def get_v1_api():
    """tweepy v1 API for media upload"""
    auth = tweepy.OAuth1UserHandler(
        os.environ.get("X_API_KEY"),
        os.environ.get("X_API_SECRET"),
        os.environ.get("X_ACCESS_TOKEN"),
        os.environ.get("X_ACCESS_SECRET"),
    )
    return tweepy.API(auth)

def post_tweet(text, image_path=None):
    client = get_client()
    v1 = get_v1_api()
    
    media_id = None
    
    if image_path and os.path.exists(image_path):
        print(f"📸 Uploading image: {image_path}")
        media_id = v1.media_upload(image_path).media_id
        print(f"   Media ID: {media_id}")
    
    print(f"📝 Posting tweet ({len(text)} chars)...")
    resp = client.create_tweet(text=text, media_ids=[media_id] if media_id else None)
    
    if "data" in resp.data:
        tweet_id = resp.data["id"]
        print(f"✅ Tweet posted: https://x.com/VotiveAstrology/status/{tweet_id}")
        return tweet_id
    else:
        print(f"❌ Error: {resp.data}")
        return None

if __name__ == "__main__":
    if len(sys.argv) < 2:
        print("Usage: python3 x_post.py 'message' [image_path]")
        sys.exit(1)
    
    text = sys.argv[1]
    image_path = sys.argv[2] if len(sys.argv) > 2 else None
    post_tweet(text, image_path)
